### DISCUSSION TITLE
Python Debugging is too slow.

### DISCUSSION THREAD

**Discussion by jeanne150** [microsoft/debugpy#1635] [General]
When debugging with the same piece of code, the time it takes to run the result with `run` is 25.895, 

and the time it takes to run the result with `debug` is 205.
```
import time
from uuid import uuid4

a=[uuid4() for i in range(20000)]
b=a[:19000]+[uuid4() for i in range(1000)]
start=time.time()
c=[i for i in a if i in b]
print(c)
print(time.time()-start)
print("debug point here")
```

**Comment by jeanne150** [comment]
Related on StackOverflow：[here](https://stackoverflow.com/questions/78774591/python-debugging-is-too-slow-how-to-make-it-the-debugging-faster)

**Comment by rchiodo** [comment]
Sorry but I don't think there's anything to be done about this. Python debugging is slow based on how it works. You can read more about it here:
https://docs.python.org/3/library/sys.html#sys.settrace

And here:
https://www.youtube.com/watch?v=QU158nGABxI

Debugpy is implementing the optimizations that Liran talks about. 

3.12 may get [better though](https://docs.python.org/3/library/sys.monitoring.html). The CPython runtime has implemented new tracing mechanisms. Debugpy hasn't really updated to using the new way yet, but it's a work [in progress](https://github.com/microsoft/debugpy/issues?q=is%3Aissue+is%3Aopen+%5Bsys.monitoring%5D)

**Comment by zhu0629** [comment]
I agree, debugpy is slower then pycharm debug on same device.
